Pandas loc incorrect type with multi-index

When using a multi-index dataframe, accessing values with loc returns another dataframe.  However, mypy seems to report this as an error expecting a series.  I extensively use multi-index dataframes and it makes usage of stubs very difficult.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
```
import pandas as pd

def main() -> None:
    idx = pd.MultiIndex.from_product(
        [["a", "b"], [1, 2]], names=["alpha", "num"]
    )

    df = pd.DataFrame({"a": range(4)}, index=idx)
    df2: pd.DataFrame = df.loc["a"]
    print(type(df))
    print(df)
    print(type(df2))
    print(df2)
    # reveal_locals()


if __name__ == "__main__":
    main()
```
3. Indicate which type checker you are using (`mypy` or  `pyright`).
mypy
4. Show the error message received from that type checker while checking your example.
```
demo.py:13: error: Incompatible types in assignment (expression has type "Series[Any]", variable has type "DataFrame")  [assignment]
```

**Please complete the following information:**
 - OS: [e.g. Windows, Linux, MacOS]
 Linux
 - OS Version [e.g. 22]
 CentOS Linux release 7.8.2003 (Core)
 - python version
 Python 3.8
 - version of type checker
 mypy 1.4.1 (compiled: yes)
 - version of installed `pandas-stubs`
pandas-stubs=2.0.1.230501=pyhd8ed1ab_0

**Additional context**
Add any other context about the problem here.

```
$ python demo.py 
<class 'pandas.core.frame.DataFrame'>
           a
alpha num   
a     1    0
      2    1
      3    2
b     1    3
      2    4
      3    5
c     1    6
      2    7
      3    8
<class 'pandas.core.frame.DataFrame'>
     a
num   
1    0
2    1
3    2

Process finished with exit code 0
```

--- a/pandas-stubs/core/frame.pyi
+++ b/pandas-stubs/core/frame.pyi
@@ -157,6 +157,8 @@ class _iLocIndexerFrame(_iLocIndexer):
 
 class _LocIndexerFrame(_LocIndexer):
     @overload
+    def __getitem__(self, idx: Scalar) -> Series | DataFrame: ...
+    @overload
     def __getitem__(
         self,
         idx: IndexType
@@ -184,8 +186,7 @@ class _LocIndexerFrame(_LocIndexer):
     @overload
     def __getitem__(
         self,
-        idx: ScalarT
-        | Callable[[DataFrame], ScalarT]
+        idx: Callable[[DataFrame], ScalarT]
         | tuple[
             IndexType
             | MaskType

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -2199,16 +2199,16 @@ def test_frame_scalars_slice() -> None:
 
     # Note: bool_ cannot be tested since the index is object and pandas does not
     # support boolean access using loc except when the index is boolean
-    check(assert_type(df.loc[str_], pd.Series), pd.Series)
-    check(assert_type(df.loc[bytes_], pd.Series), pd.Series)
-    check(assert_type(df.loc[date], pd.Series), pd.Series)
-    check(assert_type(df.loc[datetime_], pd.Series), pd.Series)
-    check(assert_type(df.loc[timedelta], pd.Series), pd.Series)
-    check(assert_type(df.loc[int_], pd.Series), pd.Series)
-    check(assert_type(df.loc[float_], pd.Series), pd.Series)
-    check(assert_type(df.loc[complex_], pd.Series), pd.Series)
-    check(assert_type(df.loc[timestamp], pd.Series), pd.Series)
-    check(assert_type(df.loc[pd_timedelta], pd.Series), pd.Series)
+    check(assert_type(df.loc[str_], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[bytes_], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[date], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[datetime_], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[timedelta], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[int_], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[float_], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[complex_], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[timestamp], Union[pd.Series, pd.DataFrame]), pd.Series)
+    check(assert_type(df.loc[pd_timedelta], Union[pd.Series, pd.DataFrame]), pd.Series)
     check(assert_type(df.loc[none], pd.Series), pd.Series)
 
     check(assert_type(df.loc[:, str_], pd.Series), pd.Series)
@@ -2223,11 +2223,20 @@ def test_frame_scalars_slice() -> None:
     check(assert_type(df.loc[:, pd_timedelta], pd.Series), pd.Series)
     check(assert_type(df.loc[:, none], pd.Series), pd.Series)
 
+    # GH749
+
+    multi_idx = pd.MultiIndex.from_product([["a", "b"], [1, 2]], names=["alpha", "num"])
+    df2 = pd.DataFrame({"col1": range(4)}, index=multi_idx)
+    check(assert_type(df2.loc[str_], Union[pd.Series, pd.DataFrame]), pd.DataFrame)
+
+    df3 = pd.DataFrame({"x": range(2)}, index=pd.Index(["a", "b"]))
+    check(assert_type(df3.loc[str_], Union[pd.Series, pd.DataFrame]), pd.Series)
+
 
 def test_boolean_loc() -> None:
     # Booleans can only be used in loc when the index is boolean
     df = pd.DataFrame([[0, 1], [1, 0]], columns=[True, False], index=[True, False])
-    check(assert_type(df.loc[True], pd.Series), pd.Series)
+    check(assert_type(df.loc[True], Union[pd.Series, pd.DataFrame]), pd.Series)
     check(assert_type(df.loc[:, False], pd.Series), pd.Series)
 
 

